File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
function solution(n, k) {
    const isPrime = (N) => {
        if (["1", ""].includes(N)) return false;

        for (let i = 2; i <= Math.sqrt(N); i++) {
            if (N % i === 0) return false;
        }
        return true;
    }
    return n.toString(k).split(0).filter((e) => isPrime(e)).length;
}

File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

bool IsPrime(long param)
{
    for (int i = 2; i <= sqrt(param); ++i)
        if (!(param % i)) return false;
    return (param > 1);
}

int solution(int n, int k) {
    int answer = 0;

    string str = "";
    while (n)
    {
        int temp = n % k;
        if (temp) str = to_string(temp) + str;
        else if (!str.empty()) answer += IsPrime(stol(str)) ? 1 : 0, str.clear();
        
        n /= k;
    }

    if (!str.empty()) answer += IsPrime(stol(str)) ? 1 : 0; // 마지막까지 남은 str
    return answer;
}
